### DISCUSSION TITLE
Dictionary not being generated for fprime-gds?

### DISCUSSION THREAD

**Discussion by ebenezerarunkumar** [nasa/fprime#1569] [Q&A] [answered]
Hi,

I tried to run the GDS for one of my deployments and I got the following error:
`[ERROR] dictionary location /Users/earunkumar/fprime-devel/visors-software/hsa/build-artifacts/Darwin/dict does not exist`. I have of course run `fprime-util generate` and `fprime-util build`. 

The `build-artifacts/Darwin` folder exists, but only contains the `bin` and `lib` folder, no `dict` folder. I checked a few of my other deployments and found that the same thing was happening - no dictionary was being created at this location. The dictionary CAN be found in the `Top/` folder of the deployment within the build folder(i.e for my case `hsa/build-fprime-automatic-native/hsa/Top/HsaTopologyAppDictionary.xml`).

I am currently running fprime v3.1. Does anyone have any idea why this could be happening?

**Comment by ThibFrgsGmz** [comment]
This error looks familiar, I've encountered something like this before.
It was due to the fact that the name of my CMake project did not coincide with the name of the folder or the suffix of the files in Top/.

It seems to me that other people have seen this problem and that an issue was created where the user had to be told to pay attention to the names.

**Reply by ThibFrgsGmz** [reply]
Look at this thread, this is the problem I had.

https://github.com/nasa/fprime/discussions/1450

**Reply by ebenezerarunkumar** [reply]
AHH Looks like i didn't look through the discussions well enough. This fixed my problem, thank you!

**Comment by ebenezerarunkumar** [comment, marked as answer]
See #1450 for answer
